lbplanner/services/user/register_user.php

```diff
@@ -76,32 +105,34 @@ public static function register_user($userid, $lang, $theme) {
         notifications_helper::notify_user($userid, -1, notifications_helper::TRIGGER_USER_REGISTERED);
 
         return array(
-            'userid' => $user->userid,
-            'username' => $mdluser->username,
-            'firstname' => $mdluser->firstname,
-            'lastname' => $mdluser->lastname,
+            'userid' => $lbplanneruser->userid,
+            'username' => $USER->username,
+            'firstname' => $USER->firstname,
+            'lastname' => $USER->lastname,
             'capabilities' => user_helper::get_user_capability_bitmask($userid),
-            'theme' => $user->theme,
-            'lang' => $user->language,
-            'profileimageurl' => $mdluser->profileimageurl,
+            'theme' => $lbplanneruser->theme,
+            'lang' => $lbplanneruser->language,
+            'profileimageurl' => user_helper::get_mdl_user_picture($userid),
             'planid' => $planid,
-            'colorblindness' => $user->colorblindness,
+            'colorblindness' => $lbplanneruser->colorblindness,
+            'displaytaskcount' => $lbplanneruser->displaytaskcount
         );
     }
 
-    public static function register_user_returns() {
+    public static function register_user_returns(): external_single_structure {
         return new external_single_structure(
             array(
-                'userid' => new external_value(PARAM_INT, 'The id of the user'),
+                'userid' => new external_value(PARAM_INT, 'The Moodle id of the user'),
```

Comment:
do we have our own id's for users? if not annotating the ID as 'Moodle id' is unnecessary and could cause confusion